src/fetch.js

```diff
@@ -3,52 +3,47 @@
 const Undici = require('undici')
 const Request = require('./request')
 const Response = require('./response')
+const { createUndiciRequestOptions } = require('./utils')
 const { STATUS_CODES } = require('http')
+const { promisify } = require('util')
+
+const requestAsync = promisify(Undici.Pool.prototype.request)
 
 function buildFetch (undiciPoolOpts) {
   if (arguments.length > 0) {
-    throw Error('Did you forget to build the instance? Try: `const fetch = require(\'fetch\')()`')
+    throw Error('Did you forget to build the instance? Try: `const fetch = require(\'undici-fetch\')()`')
   }
 
   const clientMap = new Map()
 
   function fetch (resource, init = {}) {
     const request = new Request(resource, init)
 
-    const origin = request.url.origin
-    let client = clientMap.get(origin)
-    if (client === undefined) {
+    const { origin } = request.url
+    let client
+    if (clientMap.has(origin)) {
```

Comment:
Revert this change, the existing code only has to perform a map lookup once

Reply:
I was always told that decent collections perform better with recently looked up values, and a `has` immediately followed by a `get` of the same key is therefore equally fast, while making the code a little more human-readable. But for lack of a benchmark, sure, I'll revert it.

Reply:
Yeah you're probably right, and honestly its going to be very minimal perf difference, but my extra rule of thumb is if its trivially easy to avoid extra method calls then I do so; thus I never even have to concern myself with "decent collections perform better with recently looked up values".

Reply:
a5e0c62